src/matcha_ml/services/analytics_service.py

```diff
@@ -80,59 +171,32 @@ def inner(*args: Any, **kwargs: Any) -> Any:
                 Any: the result of the wrapped function.
             """
             global_params = GlobalParameters()
-            result, error_code, te, ts = None, None, None, None
+
+            # tests if the tracking event name is in the enums
             if event_name.value not in {event.value for event in AnalyticsEvent}:
                 warn("Event not recognized by analytics service.")
 
+            # do the tracking
             if not global_params.analytics_opt_out:
+                result, error_code, ts, te = None, None, None, None
+                matcha_state_uuid = None
+
                 if event_name.value in [event_name.PROVISION, event_name.GET]:
-                    ts = perf_counter()
-                    result, error_code = execute_analytics_event(func, *args, **kwargs)
-                    te = perf_counter()
-
-                try:
-                    matcha_state_service = MatchaStateService()
-                except Exception as e:
-                    matcha_state_service = None
-                    error_code = e
-
-                # Get the matcha.state UUID if it exists
-                matcha_state_uuid: Optional[str] = None
-                if matcha_state_service and matcha_state_service.state_exists():
-                    try:
-                        state_id_component = matcha_state_service.get_component("id")
-                    except MatchaError:
-                        state_id_component = None
-
-                    if state_id_component is not None:
-                        matcha_state_uuid = state_id_component.find_property(
-                            property_name="matcha_uuid"
-                        ).value
-
-                        try:
-                            _check_uuid(str(matcha_state_uuid))
-                        except MatchaError as err:
-                            raise err
-
-                if event_name.value in [event_name.DESTROY]:
-                    ts = perf_counter()
-                    result, error_code = execute_analytics_event(func, *args, **kwargs)
-                    te = perf_counter()
-
-                client = analytics.Client(WRITE_KEY, max_retries=1, debug=False)
-
-                client.track(
-                    global_params.user_id,
-                    event_name.value,
-                    {
-                        "time_taken": float(te) - float(ts),  # type: ignore
-                        "error_type": f"{error_code.__class__}.{error_code.__class__.__name__}",
-                        "command_succeeded": error_code is None,
-                        "matcha_state_uuid": matcha_state_uuid,
-                    },
+                    result, error_code, ts, te = _time_event(func, *args, **kwargs)
+                    matcha_state_uuid = _get_state_uuid()
+                elif event_name.value in [event_name.DESTROY]:
```

Comment:
Isn't the lack of `matcha_state_uuid` in the context of a `DESTROY` event going to cause use to not know what environment was destroyed when we look at the analytics?

Reply:
Good spot - I added that if/else (rather than two separate ifs) but that's the reason it was originally two separate ifs. I've updated.